cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/RestartReason.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import java.util.Objects;
+
+public class RestartReason {
```

Comment:
I don't think we need this class now that we're not using the hack of always passing `BrokerConfigChange`. This means the `Function<Pod, List<RestartReason>>` could revert to `Function<Pod, List<String>>` or `Function<Pod, String>`.